### PR TITLE
Gj 2808

### PR DESCRIPTION
**Pull request by henrivuorinen**
Created atomi CheckVariableContainsValue to evaluate if varValue contains value given to analyzer.

### CODE REVIEW

**Comment by angleto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Coverage increased per file
===========================
+ src/main/scala/com/getjenny/starchat/analyzer/atoms/CheckVariableValue.scala  3
         

Complexity increasing per file
==============================
- src/main/scala/com/getjenny/starchat/analyzer/atoms/CheckVariableContainsValue.scala  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981&bid=20586248)

File: src/main/scala/com/getjenny/starchat/analyzer/atoms/StarchatFactoryAtomic.scala [hunk 1]
```diff
@@ -22,6 +22,7 @@ class StarchatFactoryAtomic extends AtomicFactoryTrait[List[String], AbstractAto
     "checkTimestamp",
     "checkTimestampVariable",
     "checkVariableValue",
+    "checkVariableContainsValue",
```
Comment (angleto) [inline src/main/scala/com/getjenny/starchat/analyzer/atoms/StarchatFactoryAtomic.scala:25]: Codacy found an issue: [The string literal 'checkVariableContainsValue' appears 2 times in the file.](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981)

File: src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import com.getjenny.analyzer.expressions.AnalyzersDataInternal
+import org.scalatest._
+import org.scalatest.wordspec.AnyWordSpec
+
+class AtomsTest extends AnyWordSpec with Matchers with ScalatestRouteTest {
+
+  val restrictedArgs = Map.empty[String, String]
+
+  val analyzerData = Map(
+    "variable1" -> "value1",
```
Comment (angleto) [inline src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala:18]: Codacy found an issue: [The string literal 'value1' appears 3 times in the file.](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981)

File: src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "variable2" -> "value2",
+  )
+
+  "checkVariableContainsValue" should {
+    "return score 1 when variable contains substring" in {
+      val k = new CheckVariableContainsValue(List("variable1", "val"), restrictedArgs)
+      k.evaluate("whatever", AnalyzersDataInternal(extractedVariables = analyzerData)).score should be(1.0)
```
Comment (angleto) [inline src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala:25]: Codacy found an issue: [The string literal 'whatever' appears 5 times in the file.](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981)

File: src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  val restrictedArgs = Map.empty[String, String]
+
+  val analyzerData = Map(
+    "variable1" -> "value1",
+    "variable2" -> "value2",
+  )
+
+  "checkVariableContainsValue" should {
+    "return score 1 when variable contains substring" in {
+      val k = new CheckVariableContainsValue(List("variable1", "val"), restrictedArgs)
```
Comment (angleto) [inline src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala:24]: Codacy found an issue: [The string literal 'val' appears 2 times in the file.](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981)

**Review by bmichele — CHANGES_REQUESTED** [review]
As it is now, the `checkVariableContainsValue` atom returns score 1 when `variableName` is not set. Please fix this and add the relevant test.

**Comment by angleto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/main/scala/com/getjenny/starchat/analyzer/atoms/CheckVariableContainsValue.scala  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981&bid=20586248)

File: src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import com.getjenny.analyzer.expressions.AnalyzersDataInternal
+import org.scalatest._
+import org.scalatest.wordspec.AnyWordSpec
+
+class AtomsTest extends AnyWordSpec with Matchers with ScalatestRouteTest {
+
+  val restrictedArgs = Map.empty[String, String]
+
+  val analyzerData = Map(
+    "variable1" -> "value1",
```
Comment (angleto) [inline src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala:18]: Codacy found an issue: [The string literal 'value1' appears 3 times in the file.](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981)

File: src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    "variable2" -> "value2",
+  )
+
+  val query = "whatever"
+
+  "checkVariableContainsValue" should {
+    "return score 1 when variable contains substring" in {
+      val k = new CheckVariableContainsValue(List("variable1", "val"), restrictedArgs)
```
Comment (angleto) [inline src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala:26]: Codacy found an issue: [The string literal 'val' appears 2 times in the file.](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981)

File: src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+import org.scalatest._
+import org.scalatest.wordspec.AnyWordSpec
+
+class AtomsTest extends AnyWordSpec with Matchers with ScalatestRouteTest {
+
+  val restrictedArgs = Map.empty[String, String]
+
+  val analyzerData = Map(
+    "variable1" -> "value1",
+    "variable2" -> "value2",
```
Comment (angleto) [inline src/test/scala/com/getjenny/starchat/analyzer/atoms/AtomsTest.scala:19]: Codacy found an issue: [The string literal 'value2' appears 2 times in the file.](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981)

File: src/main/scala/com/getjenny/starchat/analyzer/atoms/StarchatFactoryAtomic.scala (hunk 1 above)
Comment (angleto) [inline src/main/scala/com/getjenny/starchat/analyzer/atoms/StarchatFactoryAtomic.scala:25]: Codacy found an issue: [The string literal 'checkVariableContainsValue' appears 2 times in the file.](https://app.codacy.com/gh/GetJenny/starchat/pullRequest?prid=6336981)

**Comment by henrivuorinen** [comment]
> As it is now, the `checkVariableContainsValue` atom returns score 1 when `variableName` is not set. Please fix this and add the relevant test.

Latest commit has a fix to this issue, and all the tests are passing successfully

**Review by bmichele — APPROVED** [review]
Looks good to me!

**Pull request merged**